support-frontend/assets/components/page/offerStrapline.tsx

```diff
@@ -35,56 +52,36 @@ const offerStraplineStyles = (
 			: ''}
 	}
 	${from.tablet} {
-		${textSansBold20};
-		padding: 4px 20px 12px;
 		max-width: 50%;
 	}
-	${from.desktop} {
-		${textSansBold24};
-	}
+	${fontSizes[size]}
 `;
 
-type PropTypes = {
-	fgCol: string;
-	bgCol: string;
-	copy?: string;
-	orderIsAGift?: boolean;
+export const preventWidow = (text: string): string => {
+	const trimmed = text.trim();
+	const words = trimmed.split(' ');
+
+	if (words.length <= 1) {
+		return text;
+	}
+
+	const lastWord = words.pop();
+	return `${words.join(' ')}\u00A0${lastWord}`; // \u00A0 is non-breaking space
 };
 
-function OfferStrapline({
-	fgCol,
-	bgCol,
+export default function OfferStrapline({
 	copy,
-	orderIsAGift,
-}: PropTypes): ReactElement {
-	// Requirement: last line must include a minimum of 2 words
-	const noWidowWord = (c: string) => {
-		const trimmedCopy = c.trim();
-		const copyLength = trimmedCopy.length;
-		const wordArray: string[] = trimmedCopy.split(' ');
-		if (wordArray.length > 1) {
-			const lastWord: string | undefined = wordArray.pop();
-			return (
-				<div css={offerStraplineStyles(copyLength > 32, bgCol, fgCol)}>
-					<span>
-						{wordArray.join(' ')}&nbsp;{lastWord}
-					</span>
-				</div>
-			);
-		}
-		return <div css={offerStraplineStyles(false, bgCol, fgCol)}>{c}</div>;
-	};
+	size = 'medium',
+	cssOverrides,
+}: {
+	copy: string;
+	size?: Size;
```

Comment:
Can see this component used quite a few places so switch seems needed to me. Nice neat implementation of `fontSizes` I thought too.